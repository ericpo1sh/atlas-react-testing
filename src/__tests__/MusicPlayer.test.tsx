import { render, screen } from "@testing-library/react";
import MusicPlayer from "../MusicPlayer";
import CurrentlyPlaying from "../components/CurrentlyPlaying";
import Playlist from "../components/Playlist";
import { expect, test } from "vitest";

test("MusicPlayer renders correctly", () => {
  const { container } = render(<MusicPlayer />);
  expect(container).toMatchSnapshot(); 
});

test("MusicPlayer has correct flex layout and shadow styling", () => {
  const { container } = render(<MusicPlayer />);
  const musicPlayerDiv = container.querySelector(".flex");
  expect(musicPlayerDiv).not.toBeNull();
});

test("MusicPlayer uses column layout on small screens", () => {
  const { container } = render(<MusicPlayer />);
  const musicPlayerDiv = container.querySelector(".md\\:flex-row");
  expect(musicPlayerDiv).not.toBeNull();
});

test("MusicPlayer renders CurrentlyPlaying component", () => {
  const { container } = render(<MusicPlayer />);
  const currentlyPlayingDiv = container.querySelector("div > div.p-3.flex-1"); // Select the div containing CurrentlyPlaying
  expect(currentlyPlayingDiv).not.toBeNull();
});

test("MusicPlayer applies correct border styling for Playlist section", () => {
  const { container } = render(<MusicPlayer />);
  const borderElement = container.querySelector(".md\\:border-l.border-t");
  expect(borderElement).not.toBeNull();
});
