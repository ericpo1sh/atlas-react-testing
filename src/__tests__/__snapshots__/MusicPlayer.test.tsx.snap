// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MusicPlayer renders correctly 1`] = `
<div>
  <div
    class="flex self-center mt-12 flex-col md:flex-row shadow-2xl"
  >
    <div
      class="p-3 flex-1"
    >
      <div>
        <div>
          <div>
            Not Found
          </div>
          <div>
            Not Found
          </div>
          <div>
            Not Found
          </div>
          <div
            class="flex-col flex items-center mt-6 pb-1.5"
          >
            <div
              class="w-96 flex justify-between gap-4"
            >
              <img
                alt="Volume icon"
                class="w-5"
                src="/src/assets/volume.svg"
              />
              <input
                class="w-full cursor-pointer caret-jacarta-800 bg-jacarta-800 accent-jacarta-800 rounded-lg"
                max="100"
                min="0"
                type="range"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="p-3 flex-1 relative"
    >
      <div>
        Not Found
      </div>
      <div
        class="absolute inset-0 flex md:flex-col"
      >
        <div
          class="flex-1 border-slate-300 md:border-l border-t md:border-t-0"
        />
      </div>
    </div>
  </div>
</div>
`;
